src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Helpers.cs

```diff
@@ -12,54 +12,113 @@ namespace System.Text.Json
 {
     public sealed partial class Utf8JsonWriter
     {
-        private bool HasPartialStringData => PartialStringDataLength != 0;
+        /// <summary>
+        /// Assuming that the writer is currently in a valid state, this returns true if a JSON value is allowed at the current position.
```

Comment:
What does it mean for the writer to not be in a valid state? A bug in the implementation or something different? If it's the former I think we can omit the qualification, and if it's the latter it requires clarification.